File: Java_Practice/Java/dma_program1.java
//Demonstrate difference between uninitialized & zero-initialized array
public class dma_program1 {
    public static void main(String[] args) {
        int n = 5;

        int[] uninitialized = new int[n]; // Java automatically initializes to 0
        int[] zeroInitialized = new int[n]; // Same

        System.out.print("Uninitialized array values: ");
        for(int i=0;i<n;i++) System.out.print(uninitialized[i] + " ");

        System.out.print("\nZero-initialized array values: ");
        for(int i=0;i<n;i++) System.out.print(zeroInitialized[i] + " ");
    }
}

File: Java_Practice/CPP/dma_program1.cpp
//Demonstrate difference between uninitialized & zero-initialized array
#include <iostream>
using namespace std;

int main() {
    int n = 5;

    int *a = new int[n];        // Uninitialized
    int *b = new int[n]();      // Zero-initialized

    cout << "Uninitialized array: ";
    for(int i=0;i<n;i++) cout << a[i] << " ";

    cout << "\nZero-initialized array: ";
    for(int i=0;i<n;i++) cout << b[i] << " ";

    delete[] a;
    delete[] b;
    return 0;
}
